client/lib/domains/check-inbound-transfer-status.js

```diff
@@ -6,12 +6,12 @@ export function checkInboundTransferStatus( domainName, onComplete ) {
 		return;
 	}
 
-	wpcom.undocumented().getInboundTransferStatus( domainName, function ( serverError, result ) {
-		if ( serverError ) {
-			onComplete( serverError.error );
-			return;
-		}
-
-		onComplete( null, result );
-	} );
+	wpcom.req
+		.get( `/domains/${ encodeURIComponent( domainName ) }/inbound-transfer-status` )
+		.then( ( data ) => {
+			onComplete( null, data );
+		} )
+		.catch( ( error ) => {
+			onComplete( error.error );
+		} );
```

Comment:
The `checkInboundTransferStatus` function could be promisified, too, removing the `onComplete` callback. The (one) place that calls `checkInboundTransferStatus` wraps it in `new Promise` anyway.

That `checkInboundTransferStatus` call is also buggy: the function will call `onComplete( null )` when `domainName` is falsy, i.e., `error` is `null` and `result` is `undefined`, but the caller proceeds to access `result` fields without checking the value.